File: javascript/strain/strain.js
module.exports = {
  keep: function (array, filter) {
    var result = [];
    for (var i = 0; i < array.length; i++) {
      if (filter(array[i])) {
        result.push(array[i]);
      }
    }
    return result;
  },
  discard: function (array, filter) {
    var result = [];
    for (var i = 0; i < array.length; i++) {
      if (!filter(array[i])) {
        result.push(array[i]);
      }
    }
    return result;
  }
};

File: csharp/strain/Strain.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Strain
{
  public static class ExtensionMethods
  {
    public static IEnumerable<T> Keep<T>(this IEnumerable<T> strain, Func<T, bool> predicate)
    {
      return strain.Where(x => predicate(x));
    }

    public static IEnumerable<T> Discard<T>(this IEnumerable<T> strain, Func<T, bool> predicate)
    {
      return strain.Where(x => !predicate(x));
    }
  }
}